### DISCUSSION TITLE
formats.gto and movieformats.gto are not correctly made.

### DISCUSSION THREAD

**Discussion by richardssam** [AcademySoftwareFoundation/OpenRV#472] [Build Issues]
Hi, I'm building on windows with a fresh install. It all builds correctly, although it does require two times to do the first build.

However, once I launch rv.exe I find that only a very basic set of formats are supported like .sgi

The problem appears to be that the formats.gto file and movieformats.gto file are not correctly made, and end up at about 1k in size (rather than > 5K in the commercial RV format). I found that simply copying over these files from the commercial version seems to fix it. 

I havent quite figured out where these files are being built yet.

**Comment by richardssam** [comment]
I think I found what needs to be run:
cd \OpenRV\_install\bin
.\makeMovieIOformats.exe ..\PlugIns\MovieFormats\
.\makeFBIOformats.exe ..\PlugIns\ImageFormats\

I suspect its being run from the wrong directory.

**Comment by bernie-laberge** [comment]
Thanks for reporting this @richardssam,

I'll try to reproduce this issue later today with an Open RV build from scratch.
I'll report back.

Note that as you probably know, Open RV has fewer supported movie formats than the commercial RV when building OpenRV without any modifications. That's because only the royalty free codecs are included by default in Open RV:
https://github.com/AcademySoftwareFoundation/OpenRV/blob/950ec03afeefa4a0ead147adae8ec685fbee9753/cmake/dependencies/ffmpeg.cmake#L197
But still, the movieformats.gto should be around 12KB (as opposed to 14KB for the commercial RV).
I will look into this and report back. Thanks!

**Comment by richardssam** [comment]
What was generated seemed to be 1k.

Just out of curiosity, is this file still necessary? And/or would it make
sense as json or yaml? gto isnt a terribly common format ;)

Sam

**Reply by bernie-laberge** [reply]
I am not sure why. My guess is that the original RV creators decided to use such a file for security purpose or something to make sure that only the codecs appearing in the list were loaded?
It also doesn't help that the content of the file is compressed (Writer::CompressedGTO)

To debug it in the past, I have often changed it from Writer::CompressedGTO to Writer::TextGTO here:
https://github.com/AcademySoftwareFoundation/OpenRV/blob/0ba6a733689557cc42b925c4bbefcae02fa9d07d/src/bin/imgtools/makeMovieIOformats/main.cpp#L98

**Comment by bernie-laberge** [comment]
FYI: I just rebuilt from scratch the latest Open RV [commit](https://github.com/AcademySoftwareFoundation/OpenRV/commit/0ba6a733689557cc42b925c4bbefcae02fa9d07d) on my Windows system and it built without errors. (Typical it works on my system answer sorry!).
When I get the chance this week, I will use a brand new windows machine and reinstall everything from scratch. Sometimes MSYS2 gets updated and it can cause issues with the Open RV build. You might be experiencing that...

**Comment by richardssam** [comment]
This was the only error I got, so the build process is definitely getting
much better. Also this error is also pretty easy to recover from, once you
know the trick.

Thanks again for looking into it.

Sam
